lib/exercism/user.rb

```diff
@@ -140,26 +140,30 @@ def items_where(table, condition)
 
   def five_a_day_exercises_sql
     <<-SQL
-      SELECT * FROM (
-        SELECT DISTINCT ON (s.user_id)
-          a.user_id AS commenter, s.user_id AS ex_author_id, u.username AS ex_author, s.language, s.slug, s.nit_count, s.key, ue.last_activity_at
-          FROM acls a
-          INNER JOIN submissions s
-            ON s.language = a.language AND s.slug = a.slug
-          INNER JOIN user_exercises ue
-            ON s.user_exercise_id = ue.id
-          INNER JOIN users u
-            ON s.user_id = u.id
-          INNER JOIN comments c
-            ON s.id = c.submission_id
-          WHERE a.language = s.language
-          AND a.slug = s.slug
-          AND a.user_id <> c.user_id
-          AND a.user_id <> s.user_id
+      SELECT DISTINCT
+        ON (e.user_id)
+        a.user_id AS commenter,
+        e.user_id AS ex_author_id,
+        u.username AS ex_author,
+        a.language, a.slug, s.key, s.nit_count
```

Comment:
We don't show it, but we use it to sort the submissions to return the ones with the lowest nit count.

Both Mixolidia and I are having trouble with redirects using e.key. We are getting "Exercise not found" when using `e.key` instead of `s.key` and `/exercises/:key` instead of `/submissions/:key`.